### PR TITLE
[ios/mac] Fixes potential AL_INVALID_OPERATION while stopping a long sound.

### PR DESCRIPTION
**Pull request by dumganhar**

### CODE REVIEW

File: cocos/audio/apple/AudioPlayer.mm [hunk 1]
```diff
@@ -125,10 +132,19 @@ of this software and associated documentation files (the "Software"), to deal
...
+    ALOGVV("Before alSourceStop, id=%u", _id);
+    CHECK_AL_ERROR_DEBUG();
+    alSourceStop(_alSource);
+    CHECK_AL_ERROR_DEBUG();
+    ALOGVV("Before alSourcei, id=%u", _id);
+
+    // Note: buffers have to be detached before calling 'alDeleteBuffers',
+    // otherwise 'alDeleteBuffers' will cause AL_INVALID_OPERATION.
+    // Even if it's a streaming source, there's no need to screw around with alSourceUnqueueBuffers.
+    alSourcei(_alSource, AL_BUFFER, 0);
```
Comment (dumganhar) [inline cocos/audio/apple/AudioPlayer.mm]: This line fixed the problem, other changes only make our code safer and stronger.

**Comment by dumganhar** [comment]
@zilongshanren , please review.

**Comment by minggo** [comment]
@dumganhar could you please connect the corresponding issue related with this PR if have any one.

**Comment by dumganhar** [comment]
This pr wasn't merged, actually it didn't resolve any issue, so I closed it.

**Pull request closed without merging**